### PR TITLE
Bug fixes related to Members

### PR DESCRIPTION
**Pull request by anuragP04**
Fixed the following issues:

While creating test, if description is entered and then removed then user is able to click on Next, but ideally it should not be
Show error messages while adding member
When section’s page is empty, it should not show side section
Fixed the variable names' issue in cypress test cases

### CODE REVIEW

File: app/components/members/MembersHeader.tsx [hunk 1]
```diff
@@ -5,33 +5,34 @@ import { useLoaderData } from '@remix-run/react'
...
-          </button>
-        </div>
+    <>
+      <div className="flex items-center justify-between pb-12">
+        <h1 className="text-3xl font-bold leading-9">
+          {commonMembers.members}
+        </h1>
+        <button
+          id="addMember"
+          className="h-9 rounded-lg bg-blue-900 px-4 text-xs font-medium leading-4 text-[#F0FDF4]"
```
Comment (NitinCopods) [inline app/components/members/MembersHeader.tsx:27]: Remove hard-coded color

File: app/routes/sections.tsx [hunk 2]
```diff
@@ -139,6 +137,7 @@ export default function SectionPage() {
 
   useEffect(() => {
     if (data.sections.length && !data.selectedSectionId) {
+      console.log('1', data?.filters)
```
Comment (NitinCopods) [inline app/routes/sections.tsx]: remove this console

File: app/routes/sections.tsx [hunk 3]
```diff
@@ -153,6 +152,7 @@ export default function SectionPage() {
           [sortBy]: order,
         },
       }
+      console.log('2', data?.filters)
```
Comment (NitinCopods) [inline app/routes/sections.tsx]: remove

File: app/routes/sections.tsx [hunk 4]
```diff
@@ -166,7 +166,8 @@ export default function SectionPage() {
     if (action) {
       if (action.resp?.status === 200) {
         setShowAddSectionModal(false)
-        toast.success('Section added successfully..!')
+        toast.success(action?.resp?.title)
+        console.log('3', data?.filters)
```
Comment (NitinCopods) [inline app/routes/sections.tsx]: this one

File: app/components/members/MembersHeader.tsx [hunk 5]
```diff
@@ -5,34 +5,32 @@ import { useLoaderData } from '@remix-run/react'
...
-          >
-            + {members.addMember}
-          </button>
-        </div>
+    <>
+      <div className="flex items-center justify-between pb-12">
+        <h1 className="text-3xl font-bold leading-9 membersHeading">{members.members}</h1>
+        <button
+          id="add-member"
+          className="h-9 rounded-lg bg-blue-900 px-4 text-xs font-medium leading-4 text-[#F0FDF4]"
```
Comment (yash6102000) [inline app/components/members/MembersHeader.tsx:27]: fix this hardcoded value and also remove leading

File: app/routes/members.tsx [hunk 6]
```diff
@@ -55,25 +52,25 @@ export const action: ActionFunction = async ({ request }) => {
 
     if (typeof firstName !== 'string' || firstName.length === 0) {
       return json<ActionData>(
-        { errors: { firstName: 'firstName is required' } },
+        { errors: { title: 'Firstname is required', status: 400 } },
```
Comment (yash6102000) [inline app/routes/members.tsx:55]: it should be First Name is required and create constant for this

File: app/routes/members.tsx [hunk 7]
```diff
@@ -55,25 +52,25 @@ export const action: ActionFunction = async ({ request }) => {
...
         { status: 400 }
       )
     }
     if (typeof lastName !== 'string' || lastName.length === 0) {
       return json<ActionData>(
-        { errors: { lastName: 'lastName is required' } },
+        { errors: { title: 'Lastname is required', status: 400 } },
```
Comment (yash6102000) [inline app/routes/members.tsx:61]: it should be Last Name is required and create constant for this

File: app/routes/members.tsx [hunk 8]
```diff
@@ -55,25 +52,25 @@ export const action: ActionFunction = async ({ request }) => {
...
         { status: 400 }
       )
     }
     if (typeof email !== 'string' || email.length === 0) {
       return json<ActionData>(
-        { errors: { email: 'email is required' } },
+        { errors: { title: 'Email is required', status: 400 } },
```
Comment (yash6102000) [inline app/routes/members.tsx:67]: create constant for this

File: app/routes/members.tsx [hunk 9]
```diff
@@ -55,25 +52,25 @@ export const action: ActionFunction = async ({ request }) => {
...
         { status: 400 }
       )
     }
     if (typeof roleId !== 'string' || roleId.length === 0) {
       return json<ActionData>(
-        { errors: { roleId: 'roleId is required' } },
+        { errors: { title: 'Role is required', status: 400 } },
```
Comment (yash6102000) [inline app/routes/members.tsx:73]: create constant for this

File: app/routes/members.tsx [hunk 10]
```diff
@@ -85,16 +82,25 @@ export const action: ActionFunction = async ({ request }) => {
         addHandle = json<ActionData>(
           {
             resp: {
-              status: 'Member Added Successfully..!',
-              check: new Date(),
+              title: 'Member Added Successfully..!',
```
Comment (yash6102000) [inline app/routes/members.tsx:85]: create constant for this

File: app/routes/members.tsx [hunk 11]
```diff
@@ -85,16 +82,25 @@ export const action: ActionFunction = async ({ request }) => {
...
+              status: 200,
             },
           },
           { status: 200 }
         )
       })
       .catch((err) => {
+        let title = 'Something went wrong. Please try again.'
+        if (err.code === 'P2002') {
+          title = 'Member with this email id already exists!'
```
Comment (yash6102000) [inline app/routes/members.tsx:95]: create constant for this

File: app/routes/members.tsx [hunk 12]
```diff
@@ -85,16 +82,25 @@ export const action: ActionFunction = async ({ request }) => {
...
-              check: new Date(),
+              title: 'Member Added Successfully..!',
+              status: 200,
             },
           },
           { status: 200 }
         )
       })
       .catch((err) => {
+        let title = 'Something went wrong. Please try again.'
```
Comment (yash6102000) [inline app/routes/members.tsx:93]: create constant for this

File: app/routes/members.tsx [hunk 13]
```diff
@@ -104,49 +110,55 @@ export const action: ActionFunction = async ({ request }) => {
   if (action === 'delete') {
     if (typeof formData.get('id') !== 'string') {
       return json<ActionData>(
-        { errors: { title: 'Description is required' } },
+        { errors: { title: 'Description is required', status: 400 } },
```
Comment (yash6102000) [inline app/routes/members.tsx:113]: create constant for this

File: app/routes/members.tsx [hunk 14]
```diff
@@ -104,49 +110,55 @@ export const action: ActionFunction = async ({ request }) => {
...
         { status: 400 }
       )
     }
     let deleteHandle = null
     await deleteUserById(formData.get('id') as string)
       .then((res) => {
         deleteHandle = json<ActionData>(
-          { resp: { status: 'Deleted Successfully..!' } },
+          { resp: { title: 'Deleted Successfully..!', status: 200 } },
```
Comment (yash6102000) [inline app/routes/members.tsx:121]: create constant for this

File: app/routes/members.tsx [hunk 15]
```diff
@@ -104,49 +110,55 @@ export const action: ActionFunction = async ({ request }) => {
...
           { status: 200 }
         )
       })
       .catch((err) => {
         deleteHandle = json<ActionData>(
-          { errors: { status: err, check: new Date() } },
+          {
+            errors: {
+              title: 'Something went wrong. Please try again',
```
Comment (yash6102000) [inline app/routes/members.tsx:129]: create constant for this

File: app/routes/sections.tsx [hunk 16]
```diff
@@ -267,36 +279,43 @@ export default function SectionPage() {
...
               sortByDetails={sortByDetails}
-              actionStatusData={sectionActionData?.resp?.status}
-              err={sectionActionData?.resp?.status}
+              actionStatusData={sectionActionData?.resp?.status as string}
+              err={sectionActionData?.resp?.status as string}
             />
+            </div>
+
+            {/* section details */}
+            <div className={`z-10 flex flex-1 items-center `}>
```
Comment (yash6102000) [inline app/routes/sections.tsx:288]: remove extra spacing

File: app/components/members/MembersHeader.tsx (hunk 1 above)
Comment (Dhruv-Copods) [reply app/components/members/MembersHeader.tsx:27]: issue resolved in another branch

**Review by NitinCopods — APPROVED** [review]

**Pull request merged**